src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaWriter.java

```diff
@@ -55,12 +61,34 @@
     /** Default(UNIX) file separator. */
     private static final String DEFAULT_FILE_SEPARATOR = "/";
 
+    /** Logger for logging violations. */
+    private static final Logger LOG = Logger.getLogger(XmlMetaWriter.class.getName());
```

Comment:
sevntu complains saying 
```
(extension) AvoidModifiersForTypes: 'Logger' instance should not have 'static' modifier.
```
I can't declare `Logger` as `static` and moreover sonar cloud also suggests https://rules.sonarsource.com/java/tag/convention/RSPEC-1312

Reply:
Why are you using a logger? Checkstyle doesn't use loggers to report issues/violations to the user. If you look at other AbstractChecks, you will see it uses the `log()` method to give out violations.

I would expect is to still use that in this case.

Reply:
Yes it does, I checked those earlier also but all methods for logging are not static but all methods in `XmlMetaWriter` are static as its not permitted to create its instance, that's why I chose logger.